web/src/games/kriegspiel/Board.tsx

```diff
@@ -0,0 +1,770 @@
+          {/* svg Game Board */}
+          {gameBoard}
+        </div>
+
+        {/* info UI */}
+        <div
+          style={{
+            /* position: "fixed", right:"0", */ maxWidth: '25%',
+            float: 'left',
+            border: `2px solid ${pico8Palette.dark_green}`,
+            backgroundColor: `${pico8Palette.white}`,
+          }}
+        >
+          {G.editMode ? sideBarEdit : sideBarPlay}
+
+          <p>
+            More information <a href="https://github.com/iamcxds/kriegspiel">here</a>.{' '}
+            <input
+              type="button"
+              value="Edit Mode"
+              onClick={() => {
+                moves.setEditMode(!G.editMode);
+              }}
+            />
+          </p>
+        </div>
+      </div>
+    </main>
+  );
+};
+
+function renderLayer<T>(
+  objRender: (a: T, b: CellID) => JSX.Element | JSX.Element[],
+  objLst: readonly T[] = Array(BoardSize.mx * BoardSize.my).fill(null),
+) {
+  return objLst.map((obj, id) => {
+    const pos = CId2Pos(id);
+    return gTranslate(objRender(obj, id), pos.x, pos.y);
+  });
+}
+function renderPiece(obj: ObjInstance) {
+  return (
+    <>
+      {
+        <circle
+          cx="0.5"
+          cy="0.5"
+          r="0.4"
+          stroke={pico8Palette.dark_grey}
+          strokeWidth="0.05"
+          fill={fictionColor(obj.belong)}
+        />
+      }
+      {renderStr(obj.objRender)}
+      {!obj.supplied && renderStr('😅', 0.4)}
+      {obj.retreating && renderStr('🏃‍♂️', 0.4)}
+    </>
+  );
+}
+function renderStr(str: string, size: number = 0.5) {
+  return (
+    <text fontSize={`${size}`} x="0.5" y="0.5" dominantBaseline="middle" textAnchor="middle">
+      {str}
+    </text>
+  );
+}
+
+function gTranslate(jsx: JSX.Element | JSX.Element[], x = 0, y = 0) {
+  return <g transform={`translate(${x} ${y})`}>{jsx}</g>;
+}
+
+function drawLine(stCId: CellID, edCId: CellID, color: string = 'black', width: number = 0.1, dash: string = '') {
+  const stPos = CId2Pos(stCId);
+  const edPos = CId2Pos(edCId);
+  return (
+    <line
+      x1={stPos.x + 0.5}
+      y1={stPos.y + 0.5}
+      x2={edPos.x + 0.5}
+      y2={edPos.y + 0.5}
+      stroke={color}
+      strokeWidth={width}
+      strokeDasharray={dash}
+    />
+  );
+}
+
+function spanBGColor(jsx: JSX.Element | JSX.Element[], color: string) {
+  return <span style={{ backgroundColor: color, whiteSpace: 'normal' }}>{jsx}</span>;
+}
+
+function fictionColor(pID: P_ID) {
+  switch (pID) {
+    case '0':
+      return pico8Palette.blue;
+    case '1':
+      return pico8Palette.orange;
+  }
+}
+
+const pico8Palette = {
+  black: '#00000',
+  dark_blue: '#1d2b53',
+  dark_purple: '#7e2553',
+  dark_green: '#008751',
+  brown: '#ab5236',
+  dark_grey: '#5f574f',
+  light_grey: '#c2c3c7',
+  white: '#fff1e8',
+  red: '#ff004d',
+  orange: '#ffa300',
+  yellow: '#ffec27',
+  green: '#00e436',
+  blue: '#29adff',
+  lavender: '#83769c',
+  pink: '#ff77a8',
+  light_peach: '#ffccaa',
+};
```

Comment:
If possible, it is more readable to use CSS instead, using .module.css, here is an example: https://github.com/freeboardgames/FreeBoardGames.org/blob/b596fda05f77b5ac208e842769fcd25f6d40f665/web/src/games/zooparade/components/bdeck.tsx#L13

Reply:
I find it is easier to control by script in this way.